### DISCUSSION TITLE
grid and rows

### DISCUSSION THREAD

**Discussion by kyrianae** [grafana/grafonnet#67] [Q&A]
Hi

I didn't succeed to manage grids with rows
I want a first row with 8 panels
than after rows with 2 panels

the generation work, I see positions in the json, but only empty rows in grafana

does someone have an example of multi grid or do I have to complete position and size by hand in the jsonnet file for all my panels ?

**Comment by Duologic** [comment]
I've responded to a similar question in #52

**Comment by Duologic** [comment]
We've added a [few functions](https://grafana.github.io/grafonnet/API/util.html) that can be used to generate your own custom grid, it's a bit of a toolbox that can get you started.

These functions are now used in the original `makeGrid()`, which serves as an example on how to use it:
https://github.com/grafana/grafonnet/blob/c13ba17a4d342c0cd23ac55fde640c068a41fe6a/custom/util/grid.libsonnet#L30-L76
